src/pages/BlockInfoPage.tsx

```diff
@@ -121,7 +119,7 @@ const BlockInfoPage = () => {
     redirectToTransactionIfExists()
   }, [blockInfo, id, client, blockInfoError, navigate])
 
-  return !infoLoading && (!blockInfo || blockInfoStatus !== 200) ? (
+  return !infoLoading && (!blockInfo || (blockInfoStatus && blockInfoStatus !== 200)) ? (
```

Comment:
Nit: As we're setting `blockInfoStatus` in case of error only, we could remove `blockInfoStatus !== 200` and rename the thing `blockInfoErrorCode`. Or better yet, change the format of `blockInfoError` to be:
```typescript 
{
  message: string
  code: number
}
```

WDYT?